dev-docs/modules/topLevelPaapi.md

```diff
@@ -0,0 +1,124 @@
+module_code : topLevelPaapi
+display_name : Run top level PAAPI auctions
+enable_download : true
+sidebarType : 1
+---
+
+# Top level PAAPI module
+
+This module allows Prebid.js to support PAAPI by running on-device auctions as the top level seller.
+
+## Comparison with paapiForGpt
+
+Both this module and [paapiForGpt](/dev-docs/modules/paapiForGpt.html) allow bid adapters to participate in PAAPI auctions as component sellers.
+
+With paapiForGpt, bidders' intent to participate in PAAPI is submitted to GPT, which can then decide how to run the on-device auction. 
+With topLevelPaapi, Prebid.js directly manages the on-device auction, trading ease of use for more control.
+
+## Publisher Integration
+
+To use topLevelPaapi:
+
+- you'll need a [decision logic URL](https://github.com/WICG/turtledove/blob/main/FLEDGE.md#23-scoring-bids) that has been [attested](https://github.com/privacysandbox/attestation) with Google. How to write decision logic and how to attest it are both out of scope for this document.
+- include this module with your Prebid.js bundle; this also automatically includes the [PAAPI module](/dev-docs/modules/paapi.html)
+
+    ```bash
+   gulp build --modules=topLevelPaapi,...     
+    ```
+
+- [configure this module](#config)
+- render PAAPI bids (see [examples](#examples)) 
+
+<a id="config"></a>
+## Module Configuration
+
+This module exposes the following settings:
+
+{: .table .table-bordered .table-striped }
+|Name |Type |Description |Notes |
+| ------------ | ------------ | ------------ |------------ |
+|paapi.topLevelSeller | Object | | |
+|paapi.topLevelSeller.auctionConfig | Object | Base [AuctionConfig](https://github.com/WICG/turtledove/blob/main/FLEDGE.md#2-sellers-run-on-device-auctions) to use in `runAdAuction` | Only `seller` and `decisionLogicURL` are required |
+|paapi.topLevelSeller.autorun | Boolean | If `true` (the default) , automatically start PAAPI auctions as soon as possible | |
+|paapi.topLevelSeller.overrideWinner | Boolean | If `true`, replace contextual winners with PAAPI winners as they are rendered. Default is `false`. | see [example](#overrideWinner) |  
+
+<a id="examples"></a>
+## Examples
+
+### Basic Example using renderAd
+
+```javascript
+pbjs.setConfig({
+  paapi: {
+      enabled: true,
+      defaultForSlots: 1,
+      topLevelSeller: {
+        auctionConfig: {
+          seller: 'https://www.publisher.com',
+          decisionLogicURL: 'https://www.publisher.com/decisionLogic.js',
+        },
+      }
+  }
+})
+```
+
+With the above, `navigator.runAdAuction` is invoked once per ad unit, and the result is made available through [`getPAAPIBids`](/dev-docs/publisher-api-reference/getPAAPIBids.html):
+
+```javascript
+pbjs.requestBids({
+  bidsBackHandler: function(contextualBids) {
+    pbjs.getPAAPIBids().then(paapiBids => {
+        Object.entries(contextualBids).forEach(([adUnitCode, {bids}]) => {
+            const paapiWinner = paapiBids[adUnitCode];
```

Comment:
so paapiBids[ adUnitCode ] should have an entry only if the highest paapi bid for the adUnit has higher cpm ( or is somehow has more value) than the contextual bid. Is it a correct understanding?

Reply:
That's the desired outcome, but it depends on what happens in the PAAPI auction. The only thing you get out of it is a bid to render or nothing. 

Prebid passes a floor signal to the sandbox, the logic inside it is supposed to respect it and only bid if it can beat it. In due time we can provide some reference implementation for the decision logic that shows how to do this.

Reply:
Thank you for the explanation!